### PR TITLE
UCP/TAG: Hash tags coming to offload-capable iface

### PR DESCRIPTION
**Pull request by brminich**
## What
Hash all tags of messages arriving to offload-capable interface if more than one interface is activated on the worker.

## Why ?
This is needed to avoid posting receive buffers, which are expected to arrive from offload incapable iface (like shm), to the HW.

## How ?
Add tag to offload hash table if:
- more than 1 interface activated on the worker
- message length is bigger than TM_THRESH (to avoid ~10ns penalty for small messages, which are not supposed to be offloaded)

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7779/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@hoopoepg, please review

**Comment by brminich** [comment]
@yosefe, please review

File: src/ucp/tag/offload.c [hunk 1]
```diff
@@ -21,28 +21,28 @@ int ucp_tag_offload_iface_activate(ucp_worker_iface_t *iface)
...
 
-    if (!worker->tm.offload.num_ifaces) {
+    if (!worker->tm.offload.activated) {
         ucs_assert(worker->tm.offload.thresh       == SIZE_MAX);
         ucs_assert(worker->tm.offload.zcopy_thresh == SIZE_MAX);
         ucs_assert(worker->tm.offload.iface        == NULL);
 
         worker->tm.offload.thresh       = context->config.ext.tm_thresh;
         worker->tm.offload.zcopy_thresh = context->config.ext.tm_max_bb_size;
+        worker->tm.offload.activated    = 1;
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: can we check worker->tm.offload.iface == NULL instead of adding "activated" field?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7829/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7832/ for details (Mellanox internal link).

**Comment by brminich** [comment]
Jenkins failure

bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7839/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7843/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7846/ for details (Mellanox internal link).

**Comment by brminich** [comment]
failure is #2915 

bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7851/ for details (Mellanox internal link).

**Pull request merged**